src/Toolkit/Collection.php

```diff
@@ -961,49 +961,63 @@ public function sort(...$args): static
 		$fields = [];
 
 		foreach ($args as $arg) {
-			// get the index of the latest field array inside the $fields array
-			$currentField = $fields ? count($fields) - 1 : 0;
+			// get the index of the latest field array inside $fields
+			$field = array_key_last($fields);
 
-			// detect the type of argument
-			// sorting direction
-			$argLower = is_string($arg) ? strtolower($arg) : null;
+			// normalize $arg
+			$arg = match (true) {
+				is_string($arg) => strtolower($arg),
+				default         => $arg
+			};
```

Comment:
As much as I love the match clauses, I'm not sure I prefer it here over the old version to be honest. I found that line to be very readable. I would always use match when the optional lines get too long and complicated to read, but here it's really short and quite easy to scan.

Reply:
For me personally it's easier to read logic vertically, I think. I always have troubles with the inline syntax to spot where is the condition where does the true-value start, where the false-value. But it's not a big issue. If you and/or others prefer it, I'll turn it back. 🙈

Reply:
I've just converted the three lines back.